modules/data.atmosphere/R/download.NARR_thredds.R

```diff
@@ -0,0 +1,260 @@
+#' Retrieve NARR data using thredds
+#'
+#' @param start_date Start date for meteorology
+#' @param end_date End date for meteorology
+#' @param lat.in Latitude coordinate
+#' @param lon.in Longitude coordinate
+#' @param progress Whether or not to show a progress bar (default = `TRUE`)
+#' @return `tibble` containing time series of NARR data for the given site
+#' @author Alexey Shiklomanov
+#' @examples
+#' dat <- get_NARR_thredds("2008-01-01", "2008-01-15", 43.3724, -89.9071)
+get_NARR_thredds <- function(start_date, end_date, lat.in, lon.in,
+                             progress = TRUE) {
+
+  PEcAn.logger::severeifnot(
+    length(start_date) == 1,
+    msg = paste("Start date must be a scalar, but has length", length(start_date))
+  )
+
+  PEcAn.logger::severeifnot(
+    length(end_date) == 1,
+    msg = paste("End date must be a scalar, but has length", length(end_date))
+  )
+
+  PEcAn.logger::severeifnot(
+    length(lat.in) == 1,
+    msg = paste("Latitude must be a scalar, but has length", length(latitude))
+  )
+
+  PEcAn.logger::severeifnot(
+    length(lon.in) == 1,
+    msg = paste("Longitude must be a scalar, but has length", length(longitude))
+  )
+
+  narr_start <- lubridate::ymd("1979-01-01")
+  narr_end <- lubridate::ymd("2018-03-31")
```

Comment:
So NARR is updated pretty frequently, monthly I think. For checking end date, I'd rather it be something like if date < (current date - one month) or something like that. Otherwise, this code will prevent runs past March 2018.